lib/Sema/DerivedConformanceEquatableHashable.cpp

```diff
@@ -376,34 +1017,64 @@ deriveHashable_enum_hashValue(TypeChecker &tc, Decl *parentDecl,
   //     return index.hashValue
   //   }
   // }
+  //
+  // enum SomeEnumWithAssociatedValues {
+  //   case A, B(Int), C(String, Int)
+  //   @derived var hashValue: Int {
+  //     var result: Int
+  //     switch self {
+  //     case A:
+  //       result = _mixInt(0)
```

Comment:
Thanks!

My original implementation actually used a multiplicative hash, but I switched over to `_mixInt` once I discovered that it existed and seeing it used in the `Set<T>` implementation here: https://github.com/apple/swift/blob/5d17b31948b0ff7132149e9e86f254e69328934e/stdlib/public/core/HashedCollections.swift.gyb#L1146

Interestingly, they also start by mixing zero, which I presumed was for ensuring that the distribution isn't biased by edge cases, which in the case of `Set` would be the empty set—that's why mix the ordinals here as well.

I'm by no means a hashing expert, so I'd appreciate more input from folks who are—implementing a new hash function in stdlib and swapping out for `_mixInt` for it in this code would be straightforward, but it would be good to have some consensus on what that hash function should look like before I start down the road of an implementation 🙂